src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -56,15 +61,33 @@ def __str__(self):
         return "\n".join(map(str, self))
 
 
-def validate_role(role: str, valid_roles: List[str] = None):
+class PromptResult(str):
```

Comment:
Can we centralize those logic in this class so we can do full UT for this class?

Reply:
updated